src/common/src/catalog/schema.rs

```diff
@@ -197,6 +198,32 @@ impl Schema {
             true
         }
     }
+
+    /// Check if the schema is same as  the iceberg table schema.
+    pub fn same_as_arrow_schema(&self, arrow_schema: &ArrowSchema) -> bool {
+        if self.fields.len() != arrow_schema.fields().len() {
+            return false;
+        }
+        let mut schema_fields = HashMap::new();
+        self.fields.iter().for_each(|field| {
+            let res = schema_fields.insert(&field.name, &field.data_type);
+            // This assert is to make sure there is no duplicate field name in the schema.
+            assert!(res.is_none())
+        });
+
+        arrow_schema.fields().iter().all(|arrow_field| {
+            schema_fields
+                .get(arrow_field.name())
+                .and_then(|data_type| {
+                    if *data_type == &arrow_field.data_type().into() {
```

Comment:
This `into` currently panics on unsupported types. We need to update it to a `TryFrom` and handle those cases.

Also note that, among the well matched types listed above:
* RW and official java iceberg uses `Binary` but icelake uses `LargeBinary`
* RW and official java iceberg uses `Time64(Micro)` but icelake uses `Time32(Micro)`, likely a bug
* RW allows `Timestamp(Micro, Some(_))`, official java iceberg uses `Timestamp(Micro, Some("UTC"))`, but icelake uses `Timestamp(Micro, None)`

Reply:
I find that here should be data type convert to arrow type, so that we can convert the internal data to arrow data and sink into icelake: 
`*data_type.into() == &arrow_field.data_type()`

For data_type convert into arrow_type, only `DataType::Serial` is unsupported type, can we directly convert `DataType::Serial` to a int64.🤔